cocotb/handle.py

```diff
@@ -625,7 +637,19 @@ def setimmediatevalue(self, value):
             self._log.critical("Unsupported type for value assignment: %s (%s)", type(value), repr(value))
             raise TypeError("Unable to set simulator value with type %s" % (type(value)))
 
-        simulator.set_signal_val_binstr(self._handle, value.binstr)
+        simulator.set_signal_val_binstr(self._handle, set_action, value.binstr)
+
+    def check_for_set_action(self, value):
```

Comment:
@eric-wieser Two questions: do you see a more elegant way to do that, and have you seen the signal setting code in the hot path of profiles? (I.e. do we need to have a closer look at performance before merging.)

Reply:
Using `isinstance` checks like this is generally slower than using method dispatching, I think.

So perhaps the thing to do is:
```python
if not isinstance(value, SetAction):
    value = Deposit(value)
return value._as_gpi_args_for(self)
```
I haven't done any profiling for a while, so don't know what the hot path is